runner/src/load_generator/utils/ceramic_models_utils.rs

```diff
@@ -0,0 +1,171 @@
+use crate::scenario::ceramic::models::{RandomModelInstance, SmallModel};
+use crate::scenario::ceramic::CeramicClient;
+use anyhow::Result;
+use ceramic_http_client::{
+    api::{self},
+    ceramic_event::StreamId,
+    ModelAccountRelation, ModelDefinition,
+};
+use reqwest::Client;
+
+#[derive(Clone, Debug)]
+pub struct CeramicModelUser {
+    /**
+     * The ceramic client
+     */
+    pub ceramic_client: CeramicClient,
+    /**
+     * The http client
+     */
+    pub http_client: Client,
+    /**
+     * The base URL
+     */
+    pub base_url: Option<String>,
+}
+
+impl CeramicModelUser {
+    /**
+     * Index a model
+     *
+     * @param model_id The model to index
+     */
+    pub async fn index_model(&self, model_id: &StreamId) -> Result<()> {
+        let admin_code = self.get_admin_code().await?;
+        println!("Admin code: {:?}", admin_code);
+        let url = self
+            .build_url(self.ceramic_client.index_endpoint())
+            .await
+            .unwrap();
+        let req = self
+            .ceramic_client
+            .create_index_model_request(model_id, &admin_code)
+            .unwrap();
+        let resp = self.http_client.post(url).json(&req).send().await?;
+        if resp.status().is_success() {
+            Ok(())
+        } else {
+            Err(anyhow::anyhow!("Failed to index model"))
+        }
+    }
+
+    /**
+     * Generate a random model
+     *
+     * @return The stream id of the created model
+     */
+    pub async fn generate_random_model(&self) -> Result<StreamId, anyhow::Error> {
+        let small_model =
+            ModelDefinition::new::<SmallModel>("load_test_small_model", ModelAccountRelation::List)
+                .unwrap();
```

Comment:
can we use `?` or `map_err` throughout this file to remove the `unwrap()` calls? nicer to log an error than crash the generator

Reply:
actually, I changed my mind here. I think we unwrap calls that _should never fail_ and if they do, there's no point in running the test and wasting resources so we should just crash everything and get it over with